packages/myst-cli/src/transforms/code.ts

```diff
@@ -194,3 +194,45 @@ export function propagateBlockDataToCode(session: ISession, vfile: VFile, mdast:
     if (outputNode && !outputNode.visibility) outputNode.visibility = 'show';
   });
 }
+
+/**
+ * Because jupytext style notebooks can introduce code-cells in blocks
+ * after explicit thematic breaks, we need to lift them up to the parent
+ * to ensure that all executable code blocks are at the top level of
+ * the document.
+ *
+ * @param mdast
+ */
+export function transformLiftCodeBlocksInJupytext(mdast: GenericParent) {
```

Comment:
so this is a post process to prevent nesting of the code-cell in an extra block, when a thematic break is used in an `md` file. Doing this means the `code-cells` are all at root level, and more aligned with a notebook that originated from `ipynb`. 

This seems sensible from an mdast structure point of view, and right now is a pragmatic step as dependent code like the myst-theme expects executable cells to be top level children only, adding ui controls and hooking up compute to blocks( with code, output) at that level only. It's not setup to work with nested executable code-cells.